src/narrowdown.h

```diff
@@ -100,15 +102,24 @@ class NarrowDown
   // above the class.
   void add(const std::string& key, index_type i, const std::string& nextKey)
   {
-    if ( entries.empty() )
-    {
-      ASSERT(key, <, nextKey);
-      addEntry(key, i);
+    if (key >= nextKey) {
+      std::stringstream ss;
+      ss << "Dirent table is not properly sorted:\n";
+      ss << "  #0: " << key[0] << "/" << key.substr(1) << "\n";
+      ss << "  #1: " << nextKey[0] << "/" << nextKey.substr(1);
+      throw ZimFileFormatError(ss.str());
     }
-    else
-    {
+    if ( entries.empty() ) {
+      addEntry(key, i);
+    } else {
       const std::string pseudoKey = shortestStringInBetween(key, nextKey);
-      ASSERT(pred(entries.back(), pseudoKey), ==, true);
+      if (!pred(entries.back(), pseudoKey)) {
+        std::stringstream ss;
+        ss << "Dirent table is not properly sorted:\n";
+        ss << "  #0: " << key[0] << "/" << key.substr(1) << "\n";
+        ss << "  #1: " << nextKey[0] << "/" << nextKey.substr(1);
+        throw ZimFileFormatError(ss.str());
+      }
```

Comment:
The error message in this case is wrong, because the sorting is violated for the new key and a previously added key. In any case, if we need to keep this code here, I believe we better have a helper function for raising the exception (to avoid the duplication with the code above).